crypto/secp256k1/secp256.go

```diff
@@ -116,39 +119,54 @@ func GeneratePubKey(seckey []byte) ([]byte, error) {
 	return pubkey, nil
 }
 
+// NOTE: this function zeroes the input privkey. Callers need to
+// ensure they have a copy if they need to use the key after calling
+// this function.
 func Sign(msg []byte, seckey []byte) ([]byte, error) {
-	nonce := randentropy.GetEntropyCSPRNG(32)
+	defer zeroBytes(seckey)
```

Comment:
I think the zeroing should be done in package crypto, where the temporary slice is allocated.